lib/bootstrap/environment.rb

```diff
@@ -56,6 +57,31 @@ def pattern_path(path)
   end
 end
 
+def fetch_yml_settings(settings_path)
+  if settings = YAML.parse(IO.read(settings_path))
+    settings = settings.to_ruby
+    flat_settings_hash = LogStash::Util.flatten_hash(settings)
+    LogStash::Util.flatten_arguments(flat_settings_hash)
+  else
+    []
+  end
+end
+
+public
+def format_argv(argv)
+  # TODO deprecate these two arguments in the next major version.
+  # use -i irb or -i pry for console
+  if argv == ["irb"] || argv == ["pry"]
+    puts "Warn: option \"#{argv.first}\" is deprecated, use \"-i #{argv.first}\" or \"--interactive=#{argv.first}\" instead"
+    ["--interactive", argv.first]
+  else
+    # The Clamp library supports specifying the same argument multiple times
+    # and it keeps the last occurrence in an array. So in order for cli args
+    # to override the logstash.yml args, we can do `settings_from_yml + argv`
+    settings_from_yml = fetch_yml_settings(LogStash::Environment::SETTINGS_PATH)
+    settings_from_yml + argv
+  end
+end
```

Comment:
Theses methods are globally defined methods, shouldn't they be in their own module?